Scripts/Projectile.cs

```diff
@@ -31,39 +33,61 @@ public Projectile(float range, float speed, Vector3 direction, PackedScene proje
 
     public override void _Ready()
     {
-        Area projectileInstance = projectileScene.Instance() as Area;
-        projectileLength = projectileInstance.Scale.z;
-
+        RigidBody projectileInstance = projectileScene.Instance() as RigidBody;
+        // Area projectileInstance = projectileScene.Instance() as Area;
+        // projectileLength = projectileInstance.Scale.z;
+        projectileLength = 3;
         projectileInstance.Connect("body_entered", this, "OnBodyEnteredProjectile");
-
+        projectileInstance.GravityScale = 2;
         // To make the bullet's origin at the edge of the projectile weapon
         Vector3 weaponFrontDirection = -GlobalTransform.basis.z;
         Translation += projectileLength / 2 * weaponFrontDirection;
+        hit = false;
         AddChild(projectileInstance);
+        projectileInstance.ApplyCentralImpulse(
+            1.5f * projectileInstance.Mass * speed * weaponFrontDirection);
     }
 
     public override void _PhysicsProcess(float delta)
     {
-        float distanceToTravel = speed * delta;
-        Vector3 displacementToTravel = distanceToTravel * direction;
-        distanceTravelled += distanceToTravel;
+        RigidBody instance = GetNode<RigidBody>("Bullet");
+        Vector3 velocity = instance.LinearVelocity * new Vector3(1, 0, 1);
+        instance.AddCentralForce(0.003f * velocity.Length() * -velocity);
```

Comment:
Can make 0.003 a variable? And `velocity.Length() * -velocity` is meant to be the same as `velocity.LengthSquared() * -velocity.Normalized()`?

Reply:
yeah